Add new guide for forging node protection - Closes #962 (#963)
LiskHQ/lisk-docs: modules/ROOT/pages/guides/node-management/forging-node-protection.adoc

We will additionally assume that our trusted nodes have the following IPv4 addresses:

* `198.51.100.1`
* `203.0.113.1`
All nodes are assumed to listen to port `4000`.

.Trusted nodes configuration
[source,js]
----
{
  network: {
    whitelistedPeers: [{
      // this is our forging node
      ip: "192.0.0.1",
      port: 4000
    }],
}
----

.Forging node configuration
[source,js]
----
{
  network: {
    // Limit the maximum number of outbound connections to 2, the number of the trusted nodes
    maxOutboundConnections: 2,
    // Disallow inbound connections
    maxInboundConnections: 0,
    // Do not advertise our IPv4
    advertiseAddress: false,
    // Do not connect to any seed peer (the trusted nodes will be used, see below)
    seedPeers: [],
    // Those are our trusted nodes
    fixedPeers: [{198.51.100.1:4000, 203.0.113.1:4000}],
  }
}
----

== Firewalling

We recommend only allowing incoming SSH (`tcp/22`) connections on the forging node.
{url_ufw}[UFW^] is included by default on Ubuntu and can be used to that purpose.
An alternative example using `firehol` can also be found below.

=== UFW example

> To activate the UFW firewall with the desired configurations, run the following commands:

Reviewer: Either specify this should be done as root or add `sudo` below.

.UFW example
[source,bash]
----
ufw default deny incoming
ufw default allow outgoing
ufw allow ssh
ufw enable
ufw status verbose
----

The last command's output should be similar to:

----
Status: active
Logging: on (low)
Default: deny (incoming), allow (outgoing), disabled (routed)
New profiles: skip

To                         Action      From
--                         ------      ----
22/tcp                     ALLOW IN    Anywhere
----

Refer to {url_ubuntu_firewall}[^] for more information.

=== firehol example

[source,bash]
----
sudo apt-get --assume-yes install firehol <1>

> vim /etc/firehol/firehol.conf <2>

Reviewer: ```suggestion
sudo vim /etc/firehol/firehol.conf <2>
```

----

> <1> Installs firhol

Reviewer: ```suggestion
<1> Installs firehol
```

<2> Opens the firehol config file

Paste the following contents into the firehol config file:

.Sample `/etc/firehol/firehol.conf`
[source,bash]
----
interface eth0 wan
  policy drop
  server icmp accept
  server ssh accept
  client all accept
----

> Press kbd:[Esc] + kbd:[:] kbd:[w] kbd:[q] + kbd:[Enter] to save and quit the file.

Reviewer: ```suggestion
Press kbd:[Esc] + kbd:[:] kbd:[w] kbd:[q] + kbd:[Enter] to save the file and quit.
```

Now run the following commands:

[source,bash]
----
sudo firehol try <1>

> sed -i 's/^START_FIREHOL=.*$/START_FIREHOL=yes/' /etc/default/firehol <2>

Reviewer: ```suggestion
sudo sed -i 's/^START_FIREHOL=.*$/START_FIREHOL=yes/' /etc/default/firehol <2>
```

----

<1> Checks the configuration and then activates the firewall.

> It will also wait for confirmation that everything works as expected:
> if no confirmation is given (maybe you locked yourself out) it'll revert all changes.

Reviewer: ```suggestion
This commands will ask for confirmation as a fail-safe and will reverse all changes if none is given after 30 seconds.
```
